src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -622,6 +575,36 @@ protected internal bool IsArc()
             return false;
         }
 
+        /*
+         * Method for checking if Az.Ssh.ArcProxy module is installed
+         */
+        protected internal string FindProxyModuleInstalled()
+        {
+            var results = InvokeCommand.InvokeScript(
+                script: "(Get-module -ListAvailable -Name Az.Ssh.ArcProxy).Path");
+
+            if (results.Count == 0)
+            {
+                throw new AzPSApplicationException("Unable to find Az.Ssh.ArcProxy module installed. In order to connect to an arc server please install Az.Ssh.ArcProxy module");
+            }
+
+            foreach (var result in results)
+            {
+                string tempPath = (string)result.BaseObject;
```

Comment:
```suggestion
                if (result?.BaseObject is string tempPath) {
```

Reply:
This is interesting syntax. I didn't know it could be done that way.